### PR TITLE
 #347 - add prepareFields to View

### PR DESCRIPTION
**Pull request by mandragorn**
* prepareFields allows you to add cell state to your view w/o
  overriding prepare.
* For calculated state you can use prerender() to update values in
  the viewState and use the values in the template.

### CODE REVIEW

File: modules/View.js [hunk 1]
```diff
@@ -137,16 +138,87 @@
...
+     *
+     * Things to be careful of:
+     *   * If someGlobalCell is a property on the view than the property on the view will be used instead.
+     *   * if the prepared field item is not a string or object containing 'name' and 'value' properties, then an exception
+     *     will be thrown.
+     *   * 'model' and 'view' are reserved field names and cannot be reused.
+     *
+     * @return {Object} context composed of { modelName: model.toJSON() } for every model identified.
+     */
+    __getPrepareFieldsContext: function() {
```
Comment (kentmw) [inline modules/View.js]: move this to the private method section of the class

File: modules/View.js [hunk 2]
```diff
@@ -137,16 +138,87 @@
...
+     * The default prepared fields are: [ { name: 'view', value: 'viewState' }, 'model' ]
+     *
+     * Prepared fields can be defined in a couple of ways:
+     *   preparedFields = [
+     *     'model',
+     *     { name: 'app', value: someGlobalCell },
+     *     'a value that does not exist on the view',
+     *     { name: 'view', value: 'viewState' },
+     *     { name: 'patientId', value: '_patientId' },
+     *     'readOnly'
```
Comment (kentmw) [inline modules/View.js]: i'd name this `nonCellObject` or `pureObject`

File: modules/View.js [hunk 3]
```diff
@@ -137,16 +138,87 @@
...
-        };
-      } else {
-        return {
-          view: this.viewState.toJSON()
-        };
+      return this.__getPrepareFieldsContext();
+    },
+
+    /**
+     * Parses the combined arrays from the defaultPrepareFields array and the prepareFields array (or function
```
Comment (kentmw) [inline modules/View.js]: Briefly give a summary of these options in the prepare method javadoc and link to this doc

File: modules/View.js [hunk 4]
```diff
@@ -137,16 +138,87 @@
...
+     *   {
+     *     model: this.model.toJSON(),
+     *     app: someGlobalCell.toJSON(),
+     *     view: this.viewState.toJSON(),
+     *     patientId: this._patientId,
+     *     readOnly: this.readOnly
+     *   }
+     *
+     * Things to be careful of:
+     *   * If someGlobalCell is a property on the view than the property on the view will be used instead.
```
Comment (kentmw) [inline modules/View.js]: then not than. Also, add quotes around it. If the view already has a field named 'someGlobalCell'...

File: modules/View.js (hunk 3 above)
Comment (mandragorn) [reply modules/View.js]: done

File: modules/View.js (hunk 2 above)
Comment (mandragorn) [reply modules/View.js]: done

File: modules/View.js (hunk 4 above)
Comment (mandragorn) [reply modules/View.js]: done

File: modules/View.js (hunk 1 above)
Comment (mandragorn) [reply modules/View.js]: done

**Comment by kentmw** [comment]
+2

**Pull request merged**